internal/command/deploy/machines.go

```diff
@@ -495,6 +504,46 @@ func (md *machineDeployment) configureLaunchInputForReleaseCommand(launchInput *
 	if _, present := launchInput.Config.Env["RELEASE_COMMAND"]; !present {
 		launchInput.Config.Env["RELEASE_COMMAND"] = "1"
 	}
+	if launchInput.Config.Guest == nil {
+		launchInput.Config.Guest = &api.MachineGuest{}
+	}
+	// The pointer dereferences around the MachineGuest logic are awkward, but it's done this way
+	// to ensure that the struct gets copied - that way, if something touches the relcmd machine,
+	// it doesn't affect the machine it stole its config from
```

Comment:
what about having a util to deep copy any type? 

A wrapper of [copier.DeepCopy](https://github.com/superfly/flyctl/blob/f92741d971911c1f542265c5e9dd696611e6e2c1/internal/machine/config.go#L65) but for generic type

Reply:
ugh i got nerd sniped